### PR TITLE
Fix for issue#71

### PR DESCRIPTION
**Pull request by bhuvanalakshmi**
Changed makeSubscription method to return Observable of type MessageInfo which contains topic and message. This will give access to the actual topic name.

### CODE REVIEW

File: jawampa-core/src/main/java/ws/wamp/jawampa/MessageInfo.java [hunk 1]
```diff
@@ -0,0 +1,29 @@
+package ws.wamp.jawampa;
+
+public class MessageInfo<T> {
```
Comment (Matthias247) [inline jawampa-core/src/main/java/ws/wamp/jawampa/MessageInfo.java]: I think MessageInfo is a little bit too generic. I would suggest something like EventInfo or EventDetails, because it will only be used for Events and not other Messages.

File: jawampa-core/src/main/java/ws/wamp/jawampa/MessageInfo.java [hunk 2]
```diff
@@ -0,0 +1,29 @@
...
+	
+	
+	T message;
```
Comment (Matthias247) [inline jawampa-core/src/main/java/ws/wamp/jawampa/MessageInfo.java]: Please make the class immutable. -> Make fields final and remove getters.
One instance will be shared by many subscribers, and immutable class means that one subscriber can't mess things up for the other subscribers.

File: jawampa-core/src/main/java/ws/wamp/jawampa/WampClient.java [hunk 3]
```diff
@@ -361,15 +363,21 @@ public void run() {
...
-            public T call(PubSubData ev) {
+            public MessageInfo<T> call(PubSubData ev) {
                 if (eventClass == null || eventClass == Void.class) {
                     // We don't need a value
                     return null;
                 }
+                String actualTopic = null;
+               
+                if(ev.details !=null){
+                	actualTopic = ev.details.get("topic").asText();
```
Comment (Matthias247) [inline jawampa-core/src/main/java/ws/wamp/jawampa/WampClient.java]: .get("topic") can return null. This must be checked

File: jawampa-core/src/main/java/ws/wamp/jawampa/WampClient.java [hunk 4]
```diff
@@ -45,8 +47,8 @@
  * directly be instantiated.
  */
 public class WampClient {
-    
-    /** Base type for all possible client states */
+
+	/** Base type for all possible client states */
```
Comment (Matthias247) [inline jawampa-core/src/main/java/ws/wamp/jawampa/WampClient.java]: unnecessary whitespace change

File: jawampa-core/src/main/java/ws/wamp/jawampa/WampClient.java [hunk 5]
```diff
@@ -333,7 +335,7 @@ public void run() {
      * of the call be Observable&lt;String&gt;.
      * @return An observable that can be used to subscribe on the topic.
      */
-    public <T> Observable<T> makeSubscription(final String topic, final Class<T> eventClass) {
+    public <T> Observable<MessageInfo<T>> makeSubscription(final String topic, final Class<T> eventClass) {
```
Comment (Matthias247) [inline jawampa-core/src/main/java/ws/wamp/jawampa/WampClient.java]: White I agree that getting the topic is helpful for some users (like you), it makes things much more complicated for other users. And we would especially break the existing API.
My proposal would be to add 2 extra overloads which allow to retrieve the MesageInfo/EventDetails. They would need a new name because they would only differentiate in the return value. Maybe add 2 makeSubscriptionWithDetails functions that behave as your version of the methods and leave the existing ones as they are.

**Comment by Matthias247** [comment]
Thanks for the contribution! I think adding some code that helps for your use case with wildcard subscriptions is valid concern. I added some comments regarding your implementation. Could you try to address these?

File: jawampa-core/src/main/java/ws/wamp/jawampa/MessageInfo.java (hunk 1 above)
Comment (bhuvanalakshmi) [inline jawampa-core/src/main/java/ws/wamp/jawampa/MessageInfo.java]: Renamed to EventDetails

File: jawampa-core/src/main/java/ws/wamp/jawampa/MessageInfo.java (hunk 2 above)
Comment (bhuvanalakshmi) [inline jawampa-core/src/main/java/ws/wamp/jawampa/MessageInfo.java]: Addressed

File: jawampa-core/src/main/java/ws/wamp/jawampa/WampClient.java (hunk 3 above)
Comment (bhuvanalakshmi) [inline jawampa-core/src/main/java/ws/wamp/jawampa/WampClient.java]: Added a not null check

File: jawampa-core/src/main/java/ws/wamp/jawampa/WampClient.java (hunk 5 above)
Comment (bhuvanalakshmi) [inline jawampa-core/src/main/java/ws/wamp/jawampa/WampClient.java]: Don't see a need to add makeSubscriptionWithDetails for this version because this is not prefix or wildcard subscriptions 
 public <T> Observable<T> makeSubscription(final String topic, final Class<T> eventClass) {
        return makeSubscription(topic, SubscriptionFlags.Exact, eventClass);
    }

**Comment by bhuvanalakshmi** [comment]
I have addressed your comments, please review and let me know

**Comment by Matthias247** [comment]
look good. thanks
